app/controllers/spree/admin/general_settings_controller.rb

```diff
@@ -23,14 +23,9 @@ def update
 
       private
 
-      def combine_available_units_params
-        available_units = []
-        params.each do |name, value|
-          if value == "1" && unit = name.match(/available_units_(.*)/)&.captures&.first
-            available_units << unit
-          end
-        end
-        params[:available_units] = available_units.join(",")
+      def merge_available_units_params
+        params[:available_units] =
+          params[:available_units].select { |unit, checked| checked == "1" }.keys.join(",")
```

Comment:
nice! I am very happy I asked, this is much better looking :clap:

Reply:
I like the fact that this is in a separate method :ok_hand:  the name of the method explains what's happening, much better than a comment.